crates/environ/src/cache/worker.rs

```diff
@@ -319,93 +321,106 @@ impl WorkerThread {
 
         trace!("Trying to recompress file: {}", path.display());
 
+        /// Transforms `<expr> ; <ident> => <expr>` into a match expression.
+        /// Allowing you to perform a single action on the error branch.  This
+        /// is most useful for triggering logging warnings, without repeating
+        /// match or `.map_err`.
+        macro_rules! unwrap_or_warn {
+            ($ex : expr ; $error_var:ident => $warn_expr: expr) => {
+                match $ex {
+                    Ok(bytes) => bytes,
+                    Err($error_var) => {
+                        $warn_expr;
+                        return;
+                    }
+                }
+            };
+        }
```

Comment:
All uses of `unwrap_or_warn` do `error => warn!(...)`. Would it be difficult to fold that into the macro?

Would it make sense to generalize the `unwrap_or` macro used elsewhere in this file and use that here?

Reply:
Well I initially decided to have it contain `error => <expr>` so that if someone wanted to change the warning logging they wouldn't need to change the macro. If merging the two macros having the same logging in both `unwrap_or` places is okay with the project then I can make that change.

Reply:
Yeah. Unless there's some code that differs that I missed, I think it makes sense to use `unwrap_or` consistently through the code here.